main.tf

```diff
@@ -151,7 +151,6 @@ resource "google_container_node_pool" "node_pools" {
   version            = local.gke_node_version
   cluster            = google_container_cluster.k8s_cluster.name
   initial_node_count = each.value.node_count_initial_per_zone
-  node_count         = each.value.node_count_current_per_zone
```

Comment:
Do not set node_count_current_per_zone via IAC as changes to it will most often be done via console.